### ISSUE TITLE
Indigo Upgrade Meta Issue II

### ISSUE THREAD

**Issue by stonier** [turtlebot/turtlebot#170]
Moving to staggered short term releases so we can get something out there on indigo sooner rather than later despite time constraints. The following is a todo:

**Short Term Tasks**
- [x] Disable create from the dependency chain (see below)
- [x] Disable map_store from the dependency chain (see below)[https://github.com/turtlebot/turtlebot_apps/issues/126]
- [x] Move `turtlebot_teleop` scripts to `turtlebot` stack so they can be shared by apps and interactions.
- [x] Re-enable IRobot Create support (hopefully a user steps in to help test).
- [x] Rqt remocon bugfix to support remote master uri interactions.
- [x] Possible incompatibility for kinect/ubuntu on the official turtlebot laptops
- [x] Turtlebot debs
- [x] Documentation testing and runthruough (replicate hydro and leave out create/android docs).
- [x] Rqt Remocon documentation instructions where applicable

**Later Release Tasks:**

As and when we are able.
- Turtlebot teleop code switch to [ros-teleop](https://github.com/ros-teleop) implementations
- Capabilities on Turtlebot
- Drop in world canvas in place of the map_store ready for android app support.
- Android apps (once rosjava on indigo is supported and world canvas is in)

**Comment by stonier** [comment]
@jihoonl can you start an issue for the kinect problem with what you currently know.

**Comment by trainman419** [comment]
Is there anything required for the create besides doing some testing and a release? I have one around here somewhere that I can dig up and test on, and I think that's better than leaving it unsupported.

I don't see any blocking issues open on the turtlebot_create repo, and I've verified that the hydro and indigo branches build on Indigo.

**Comment by trainman419** [comment]
turtlebot_msgs released into Indigo: https://github.com/ros/rosdistro/pull/6299

**Comment by trainman419** [comment]
I fired up my official Willow Garage turtlebot and verified that it works reasonably well under Indigo, with a few patches: https://github.com/turtlebot/turtlebot_create/pull/23

**Comment by jihoonl** [comment]
FYI for turtlebot_msg https://github.com/turtlebot/turtlebot_msgs/issues/2

**Comment by trainman419** [comment]
I've released the Hydro version of map_store into indigo: https://github.com/ros/rosdistro/pull/6301

**Comment by stonier** [comment]
Great austin, thanks. Only thing that probably remains for the create is to update the create gazebo plugin. There were some depracated code we had to update for kobuki gazebo as well.

**Comment by trainman419** [comment]
The relevant issue for the create gazebo plugin is: https://github.com/turtlebot/turtlebot_create_desktop/issues/7 . I'll take a look at it, but I can't make any promises. I haven't spent much time working with gazebo.

**Comment by trainman419** [comment]
I think I fixed the create gazebo plugin: https://github.com/turtlebot/turtlebot_create_desktop/pull/16

**Comment by jihoonl** [comment]
Things to release for turtlebot_apps
- [yujin_ocs](https//github.com/yujinrobot/yujin_ocs)
- [world_canvas](https://github.com/corot/world_canvas) (for turtlebot_rapps) and replace map_store.

I am preparing to release them in nextweek.

Other things, should be fine to be released.

**Comment by trainman419** [comment]
Any progress on this? I'd really like to get these released so that I can do ARM builds.

**Comment by jihoonl** [comment]
I just created a PR to release turtlebot_create. Once it is merged, I will release turtlebot and turtlebot_create_desktop.

**Comment by trainman419** [comment]
Any progress on this? I'm still waiting for the release of turtlebot_apps into Indigo.

**Comment by jihoonl** [comment]
Quite busy to catch up on turtlebot_apps issues due to the other project deadlines... I plan to follow up remaining turtleobt issues from next week. Remaining jobs before release are the following:
- Turtlebot apps(teleop, map_nav, make_a_map) launching with QT remocon
- and hopefully ros-teleop integration

**Comment by trainman419** [comment]
I'm under contract with Qualcomm to have this released into Indigo by the end of November. I'm almost a month past that deadline now.

Indigo has already been public for at least 6 months now. There are a lot of users waiting for these packages to be released into Indigo.

The Jade release is only a few months away at this point; the OSRF team is already starting to set up the build farm for Jade. Since the proposed changes still look like they're going to require some significant development effort, I think they should be pushed off into the Jade release, so that we can get these packages released into Indigo in a timely fashion.

I'm going to have a few hours of my vacation to work on this during the week. If I haven't heard back and don't have anything new to test by Wednesday, I'm going to test and release the existing packages into Indigo.

**Comment by tfoote** [comment]
I agree it would be great to get these out. i just brought up a TurtleBot and found that I didn't have turtlebot teleop apps. Also the released version seems to not have the optenv fix released as well for minimal.launch

**Comment by stonier** [comment]
Picking up the bits and pieces here - I'd hoped Jihoon would have had them all out by now, but for various reasons it didn't happen.

@trainman419 I would have been fine with you making beta releases of everything back in November, we were unaware of your deadline though. 

Which gets to the point - let's wrap things up with some certainty instead of vague promises. @jihoonl has this week and next week to work on the non-android debs release (including post release bug fixes). 

**Initial Core Releases**

To whack the current system into shape seems pretty trivial - he will make sure `turtlebot_apps` and `turtlebot_interactions` (prev. `turtlebot_viz`) are tested and released by Wed. Korean time.

**Peripherals**

Austin, would you like to cover anything that might be missing at that point? You probably can check the create and simulator debs. And of course bugfixing.

**App Management Fixes**

Jihoon will look at a couple of minor fixes to make sure upgrades to the app manager and user side interactions runs smoothly early next week.

**Documentation**

I'll try and organise a doc'athon late next week or early the following week.

**Factory Release**

Jihoonl will use January to wrap up updates for our indigo release at the factory with the reference netbook. Major hurdle here is the failure of kinect on that reference netbook.

**Android Stuff**

At this point, I can offer no fixed date - Austin do you need a plan for this?

**Comment by stonier** [comment]
@trainman419 you have release rights on https://github.com/turtlebot-release so feel free to dive in there.

**Comment by trainman419** [comment]
@stonier admin rights on the release repos don't allow me to release the indigo-devel branches of things; for that, I need commit access to the repository itself so that I can bump the version numbers and add tags ( ie catkin_prepare_release ).

**Comment by stonier** [comment]
Ah, of course...added you to the team of core developers.

**Comment by jihoonl** [comment]
I made an alpha release today. I will test all debs and make a patch tomorrow if necessary.

**Comment by jihoonl** [comment]
I have released all turtlebot debs and start to tackle remaining issues soon.

**Comment by jihoonl** [comment]
Short term tasks are cleared . the later release tasks will be tackled with new issue #203. Closing this.
